src/working_directory.js

```diff
@@ -0,0 +1,53 @@
+import * as path from 'path';
+import { mkdirSync } from 'fs';
+
+class WorkingDirectory {
+    constructor() {
+        this.path = path.join(__dirname, '..', '..', 'packsquash');
+        mkdirSync(this.path, { recursive: true });
+    }
```

Comment:
I like how you abstracted the details of managing a working directory in its own class. Ir makes it much easier to read the code :smiling_face_with_three_hearts:. However, I think the way a temporary working directory is calculated can be improved to make it tidier and more future-proof like this:

```suggestion
import { tmpdir } from 'os';
import { mkdtempSync } from 'fs';

class WorkingDirectory {
    constructor() {
        this.path = mkdtempSync(path.join(tmpdir(), 'packsquash-'));
    }
```

Some attention should be paid to this random workspace directory when cleaning up the files generated by the action run.

Reply:
Not the way you said, fix by 23ae399067d52ae18300034310404f06ef705510.

`RUNNER_TEMP` is used in the official actions tool-cache and cache.
https://github.com/actions/toolkit/blob/4e9375da095a1d1f59b3314589fab6cd0d215139/packages/tool-cache/src/tool-cache.ts#L663